### PR TITLE
[Rule Tuning] Potential password spraying of microsoft 365 user accounts

### LINKED ISSUE #1127: [Rule Tuning] Potential Password Spraying of Microsoft 365 User Accounts
## Link to rule
https://github.com/elastic/detection-rules/blob/main/rules/microsoft-365/credential_access_microsoft_365_potential_password_spraying_attack.toml

#### Current
 - Let me know if am I wrong about this... This current rule triggers too easily and too often that I would rather turn this detection off. And there really isn't anything that a company can do about these attempt besides enforcing Multi-factor authentication for the employees.
 -  As a security engineer, I don't understand the rationale of being notified every time that a Brute Force attempt ended up being a failure anyways. 
 - This could also go 24 failure attempts and then 1 success for it be triggered.

```### 
event.dataset:o365.audit and event.provider:AzureActiveDirectory and event.category:authentication 
and event.action:UserLoginFailed and event.outcome:failure
```

#### Could potentially expand coverage by:

```
event.action:(UserLoggedIn or MailboxLogin)  
event.provider:(Exchange or AzureActiveDirectory) 
```

### PR DESCRIPTION
**Pull request by austinsonger**
## Issues

Resolves #1127 

## Summary
This can also be improved by adding another event action `PasswordLogonInitialAuthUsingPassword`

### References
- https://answers.microsoft.com/en-us/msoffice/forum/all/understanding-failed-login-in-audit-logs/3408f2ae-650f-489c-bc92-9bffbde499dd?page=2
- https://answers.microsoft.com/en-us/msoffice/forum/all/office-365-audit-logs-multiple-failed-login/80d26a71-75c7-494d-acaf-8a3bb695bcd6
- https://blog.jijitechnologies.com/get-office365-services-usage-details

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/microsoft-365/credential_access_microsoft_365_potential_password_spraying_attack.toml [hunk 1]
```diff
@@ -29,10 +29,10 @@ tags = ["Elastic", "Cloud", "Microsoft 365", "Continuous Monitoring", "SecOps",
 type = "threshold"
 
 query = '''
-event.dataset:o365.audit and event.provider:AzureActiveDirectory and event.category:authentication and event.action:UserLoginFailed and event.outcome:failure
+event.dataset:o365.audit and event.provider:(Exchange or AzureActiveDirectory) and event.category:authentication and 
+event.action:("UserLoginFailed"  or "PasswordLogonInitialAuthUsingPassword") and event.outcome:failure
```
Comment (bm11100) [inline rules/microsoft-365/credential_access_microsoft_365_potential_password_spraying_attack.toml:33]:
```suggestion
event.action:("UserLoginFailed" or "PasswordLogonInitialAuthUsingPassword") and event.outcome:failure
```

Comment (bm11100) [reply rules/microsoft-365/credential_access_microsoft_365_potential_password_spraying_attack.toml:33]: Can you bump the modified date as well?

**Review by bm11100 — COMMENTED** [review]
Do you have any sample events that use the `PasswordLogonInitialAuthUsingPassword` call?

File: rules/microsoft-365/credential_access_microsoft_365_potential_password_spraying_attack.toml [hunk 2]
```diff
@@ -3,6 +3,7 @@ creation_date = "2020/12/01"
 maturity = "production"
 updated_date = "2021/05/10"
 
+
```
Comment (bm11100) [inline rules/microsoft-365/credential_access_microsoft_365_potential_password_spraying_attack.toml:6]:
```suggestion
```

**Review by threat-punter — APPROVED** [review]

**Review by brokensound77 — APPROVED** [review]
Bump `updated_date` and LGTM 👍 

Thanks @austinsonger

**Pull request merged**